scripts/setup.sh

```diff
@@ -1,150 +1,208 @@
-		echo "Starting $tdomain"
+				# post to server
+				curl -X POST -d @/var/goddard/build.json http://hub.goddard.unicore.io/report.json?uid=$(cat /var/goddard/node.json | jq -r '.uid') --header "Content-Type:application/json"
 
-		# done
-		echo "{\"build\":\"busy\",\"process\":\"Starting $tdomain\",\"timestamp\":\"$( date +%s )\"}"  > /var/goddard/build.json
+				# write out the service file
+				sudo cat <<-EOF > /etc/nginx/conf.d/$tdomain.conf
 
-		# start the app
-		cd /var/goddard/apps/$tkey && docker run --restart=always -p $tport:8080 -d $tkey
+					server {
 
-		# done
-		echo "Adding $tdomain web server config"
+						listen                          80;
+						server_name                     $tdomain;
+						access_log                      /var/log/nginx/${tkey}.access.log;
+						error_log                       /var/log/nginx/${tkey}.error.log;
 
-		# done
-		echo "{\"build\":\"busy\",\"process\":\"Adding $tdomain web server config\",\"timestamp\":\"$( date +%s )\"}"  > /var/goddard/build.json
+						location / {
 
-		# write out the service file
-		sudo cat <<-EOF > /etc/nginx/conf.d/$tdomain.conf
+							proxy_pass                  http://127.0.0.1:${tport}\$request_uri;
+							proxy_redirect              off;
 
-			server {
+							proxy_set_header            Host             \$host;
+							proxy_set_header            X-Real-IP        \$remote_addr;
+							proxy_set_header            X-Forwarded-For  \$proxy_add_x_forwarded_for;
+							proxy_max_temp_file_size    0;
 
-				listen                          80;
-				server_name                     $tdomain;
-				access_log                      /var/log/nginx/${tkey}.access.log;
-				error_log                       /var/log/nginx/${tkey}.error.log;
+							client_max_body_size        10m;
+							client_body_buffer_size     128k;
 
-				location / {
+							proxy_connect_timeout       120;
+							proxy_send_timeout          1200;
+							proxy_read_timeout          120;
 
-					proxy_pass                  http://127.0.0.1:${tport}\$request_uri;
-					proxy_redirect              off;
+							proxy_buffer_size           128k;
+							proxy_buffers               4 256k;
+							proxy_busy_buffers_size     256k;
+							proxy_temp_file_write_size  256k;
 
-					proxy_set_header            Host             \$host;
-					proxy_set_header            X-Real-IP        \$remote_addr;
-					proxy_set_header            X-Forwarded-For  \$proxy_add_x_forwarded_for;
-					proxy_max_temp_file_size    0;
+						}
 
-					client_max_body_size        10m;
-					client_body_buffer_size     128k;
+					}
 
-					proxy_connect_timeout       120;
-					proxy_send_timeout          1200;
-					proxy_read_timeout          120;
+				EOF
 
-					proxy_buffer_size           128k;
-					proxy_buffers               4 256k;
-					proxy_busy_buffers_size     256k;
-					proxy_temp_file_write_size  256k;
+			done < /var/goddard/apps.keys.txt
 
-				}
+			# restart nginx
+			service nginx reload || true
```

Comment:
Do we want to test the nginx config with `configtest` before reloading? 
Errors here will be swallowed and will result in a misleadingly successful build message in `/var/goddard/build.json`